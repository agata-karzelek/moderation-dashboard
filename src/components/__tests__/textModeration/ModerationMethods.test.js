import React from "react";
import { shallow } from "enzyme";
import ModerationMethods from "../../textModeration/ModerationMethods";

const mockManual = {
  wordListProfanity: "true",
};

const mockAutomatic = {
  automaticProfanity: "true",
};

const mockEmpty = {
  wordListProfanity: "false",
  automaticProfanity: "false",
};

describe("Test Case for Moderation Methods", () => {
  test("Snapshot", () => {
    const wrapper = shallow(<ModerationMethods state={mockManual} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("check Automatic", () => {
    const wrapper = shallow(<ModerationMethods state={mockAutomatic} />);
    expect(wrapper.find("AutomaticProfanityMethod").exists()).toBe(true);
    expect(wrapper.find("WordListProfanityMethod").exists()).toBeFalsy();
  });

  test("check Manual", () => {
    const wrapper = shallow(<ModerationMethods state={mockManual} />);
    expect(wrapper.find("WordListProfanityMethod").exists()).toBe(true);
    expect(wrapper.find("AutomaticProfanityMethod").exists()).toBeFalsy();
  });

  test("check Empty", () => {
    const wrapper = shallow(<ModerationMethods state={mockEmpty} />);
    expect(wrapper.find("WordListProfanityMethod").exists()).toBeFalsy();
    expect(wrapper.find("AutomaticProfanityMethod").exists()).toBeFalsy();
  });
});
