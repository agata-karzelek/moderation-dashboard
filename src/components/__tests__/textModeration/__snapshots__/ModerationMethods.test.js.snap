// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Case for Moderation Methods Snapshot 1`] = `
<WordListProfanityMethod
  state={
    Object {
      "wordListProfanity": "true",
    }
  }
/>
`;
